<?xml version="1.0" encoding="UTF-8"?>
    <source package="java.rmi">
        <class name="RemoteException" line="28">
            <extends class="java.io.IOException"/>
            <comment line="52">
                                
            </comment>
            <javadoc line="28">
                A &lt;code&gt;RemoteException&lt;/code&gt; is the common superclass for a number of
                  communication-related exceptions that may occur during the execution of a
                  remote method call.  Each method of a remote interface, an interface that
                  extends &lt;code&gt;java.rmi.Remote&lt;/code&gt;, must list
                  &lt;code&gt;RemoteException&lt;/code&gt; in its throws clause.
                  &lt;p&gt;As of release 1.4, this exception has been retrofitted to conform to
                  the general purpose exception-chaining mechanism.  The &quot;wrapped remote
                  exception&quot; that may be provided at construction time and accessed via
                  the public {@link #detail} field is now known as the &lt;i&gt;cause&lt;/i&gt;, and
                  may be accessed via the {@link Throwable#getCause()} method, as well as
                  the aforementioned &quot;legacy field.&quot;
                  &lt;p&gt;Invoking the method {@link Throwable#initCause(Throwable)} on an
                  instance of &lt;code&gt;RemoteException&lt;/code&gt; always throws {@link IllegalStateException}.                
                <author>
                    Ann Wollrath                    
                </author>
                <since>
                    JDK1.1                    
                </since>
            </javadoc>
            <declaration name="serialVersionUID" type="long" line="52"/>
            <declaration name="detail" type="Throwable" line="54"/>
            <javadoc line="54">
                The cause of the remote exception.
                  &lt;p&gt;This field predates the general-purpose exception chaining facility.
                  The {@link Throwable#getCause()} method is now the preferred means of
                  obtaining this information.                
                <serial>
                                        
                </serial>
            </javadoc>
            <javadoc line="65">
                Constructs a &lt;code&gt;RemoteException&lt;/code&gt;.                
            </javadoc>
            <method name="RemoteException" type="constructor" line="68">
                <comment line="70">
                    Disallow subsequent initCause                    
                </comment>
            </method>
            <javadoc line="72">
                Constructs a &lt;code&gt;RemoteException&lt;/code&gt; with the specified
                  detail message.                
                <param>
                    s the detail message                    
                </param>
            </javadoc>
            <method name="RemoteException" type="constructor" line="78">
                <params>
                    <param name="s" type="String"/>
                </params>
                <comment line="81">
                    Disallow subsequent initCause                    
                </comment>
            </method>
            <javadoc line="83">
                Constructs a &lt;code&gt;RemoteException&lt;/code&gt; with the specified detail
                  message and cause.  This constructor sets the {@link #detail}field to the specified &lt;code&gt;Throwable&lt;/code&gt;.                
                <param>
                    s the detail message                    
                </param>
                <param>
                    cause the cause                    
                </param>
            </javadoc>
            <method name="RemoteException" type="constructor" line="91">
                <params>
                    <param name="s" type="String"/>
                    <param name="cause" type="Throwable"/>
                </params>
                <comment line="94">
                    Disallow subsequent initCause                    
                </comment>
            </method>
            <javadoc line="97">
                Returns the detail message, including the message from the cause, if
                  any, of this exception.                
                <return>
                    the detail message                    
                </return>
            </javadoc>
            <method name="getMessage" type="String" line="103">
                <scope line="104"/>
                <scope line="106"/>
            </method>
            <javadoc line="112">
                Returns the cause of this exception.  This method returns the value
                  of the {@link #detail} field.                
                <return>
                    the cause, which may be &lt;tt&gt;null&lt;/tt&gt;.                    
                </return>
                <since>
                    1.4                    
                </since>
            </javadoc>
            <method name="getCause" type="Throwable" line="119"/>
        </class>
    </source>